util/common/src/main/java/org/hibernate/search/util/common/impl/Throwables.java

```diff
@@ -72,4 +72,17 @@ public static String getFirstNonNullMessage(Throwable t) {
 		return t.getMessage();
 	}
 
+	public static String safeToString(Throwable throwableBeingHandled, Object object) {
+		if ( object == null ) {
+			return "null";
+		}
+		try {
+			return object.toString();
+		}
+		catch (Throwable t) {
+			throwableBeingHandled.addSuppressed( t );
+			return "<" + object.getClass().getSimpleName() + "#toString() threw " + t.getClass().getSimpleName() + ">";
```

Comment:
I'd rather be brief and extra safe here: we're reporting an exception during the handling of another exception... Also, the details will be in the stack trace since we used `addSuppressed` above.